### ISSUE TITLE
Locale aware index

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#383]
Hi,

When generating the index, the words that do not start with a pure ASCII letter are categorized as "symbols" and does not appear in the nearest letter.

As an example the "élément" should be in the "E". A hint to get the nearest ASCII of a non ASCII string:

```
>>> import unicodedata
>>> ut = u'élément'
>>> nt = unicodedata.normalize('NFKD', ut)
>>> nt.encode('ascii', 'ignore')
'element'
```

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/383
- Originally reported by: [glenfant](https://bitbucket.org/glenfant)
- Originally created at: 2010-04-10T20:52:10.199

**Comment by shimizukawa** [comment]
_From Éric Araujo on 2010-10-28 23:40:36+00:00_

You can use locale.strxfrm for locale-compliant comparisons instead of a lossy workaround.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2011-01-04 16:14:20+00:00_

Fixed (for now) in <<cset 5c0c465db282a6c46a8b3afe6c2c78c85db9786c>>.  Given enough time this should be done in a truly locale-dependent manner, maybe Babel supports such sorting?

**Comment by shimizukawa** [comment]
_From Anonymous on 2011-01-05 10:39:55+00:00_

Many thanks for this change. I'm impatient to update lots of french documents with the next Sphinx release.

Cheers...
